packages/realm/bindgen/src/templates/wrapper.ts

```diff
@@ -0,0 +1,405 @@
+import { TemplateContext } from "@realm/bindgen/context";
+
+import { BoundSpec, Class, Enum, Method, NamedType, Property, Struct, Type } from "@realm/bindgen/bound-model";
+import assert from "node:assert";
+import { eslintFormatter } from "../formatters";
+import { doJsPasses } from "../js-passes";
+
+function generateEnumDeclaration(e: Enum) {
+  return `export const enum ${e.jsName} { ${e.enumerators.map(({ jsName, value }) => `${jsName} = ${value}`)} };`;
+}
+
+const PRIMITIVES_MAPPING: Record<string, string | undefined> = {
```

Comment:
Because not all possible string properties result in a `string` upon lookup.

If omitted `PRIMITIVES_MAPPING["invalid"]` would have the type `string` when in fact it'll be `undefined`.